src/viewer.js

```diff
@@ -1968,7 +1972,12 @@ function openTileSource( viewer, source ) {
         _this.viewport.resetContentSize( _this.source.dimensions );
     }
 
-    _this.source.overlays = _this.source.overlays || [];
+    if( _this.preserveOverlays ){
+        _this.source.overlays = _this.currentOverlays;
```

Comment:
Looks like we can either do `_this.source.overlays` or `_this.overlays` here (see `loadOverlays()`). I think `_this.overlays` would be a little cleaner, since we wouldn't be modifying the tilesource. 

Of course the line below can stay as it originally was.